src/filecoin/FilecoinEngine.ts

```diff
@@ -373,8 +375,12 @@ export class FilecoinEngine extends CurrencyEngine<
 
   async scanTransactionsFromFilfox(
     address: string,
-    onScan: (event: { tx: EdgeTransaction; progress: number }) => void
+    onScan: (event: {
+      tx: EdgeTransaction | undefined
+      progress: number
+    }) => void
   ): Promise<void> {
+    const processedMessageCids: { [messageCid: string]: true } = {}
```

Comment:
We agreed to use a `Set<string>` for cases like this, since it's safer and theoretically faster.